server/datastore/mysql/operating_systems.go

```diff
@@ -124,3 +124,40 @@ func getHostOperatingSystemDB(ctx context.Context, tx sqlx.ExtContext, hostID ui
 	}
 	return &os, nil
 }
+
+func (ds *Datastore) CleanupHostOperatingSystems(ctx context.Context) error {
```

Comment:
I'm interested to hear perspectives on the potential for locking issues with the hosts table and how we might want to alter our approach here.